crates/runtime/src/datafusion/builder.rs

```diff
@@ -209,6 +233,52 @@ fn get_analyzer_rules() -> Vec<Arc<dyn AnalyzerRule + Send + Sync>> {
     ]
 }
 
+// This method uses unwrap_or_default, however it should never fail on the initialization. See
+// RuntimeEnv::default()
+pub(crate) fn runtime_env(
+    memory_limit: Option<u64>,
+    temp_directory: Option<String>,
+) -> Arc<RuntimeEnv> {
+    let disk_manager = if let Some(directory) = temp_directory {
+        DiskManagerConfig::new_specified(vec![directory.into()])
+    } else {
+        DiskManagerConfig::new()
+    };
+
+    let memory_pool: Arc<dyn MemoryPool> = if let Some(limit) = memory_limit {
+        let Ok(limit) = limit.try_into() else {
+            panic!("Memory limit could not fit into a usize");
```

Comment:
I think this should default to the max value if it can't fit, with an warning trace.